### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by iDVB**

### CODE REVIEW

File: src/logic/util/lambda.js [hunk 1]
```diff
@@ -1,70 +1,104 @@
 const AWS = require('aws-sdk');
+var Promise = require('bluebird');
```
Comment (simlu) [inline src/logic/util/lambda.js]: Why are you using bluebird promises? Are native AWS promises bad for some reason?

File: src/logic/util/lambda.js [hunk 2]
```diff
@@ -1,70 +1,104 @@
...
+              resultList => resultList.concat(result),
+            );
+      });
 
-  const appendLogSubscriptionInfo = functions => Promise
-    .all(functions.map(f => new Promise((resolve, reject) => cloudwatchlogs.describeSubscriptionFilters({
-      logGroupName: `/aws/lambda/${f.FunctionName}`
-    }, (err, res) => (err ? reject(err) : resolve(Object.assign(res, f)))))));
+  const appendLogRetentionInfo = functions =>
+    Promise.all(functions)
```
Comment (MacMcIrish) [inline src/logic/util/lambda.js]: This `Promise.all` does nothing at the moment, this `functions` object should be mapped to a list of promises
```suggestion
    Promise.all(functions.filter(isLogGroup).map(f => cloudWatchLogs...)
```

**Comment by iDVB** [comment]
First off, I admit I’m not an expert, so I may be wrong.
But my reasoning was that Bluebird has nice helper functions for filter and map that allow for promises. 
http://bluebirdjs.com/docs/api/promise.filter.html
This is also why `functions` is wrapped in `Promise.all()` to get the bluebird version of that filter and map functions.

**Comment by MacMcIrish** [comment]
Ah, I see. I think that the benefits of the promise filter helper method come from it predating ecma 6 type promise chaining. As the language stands at the moment, I don't know that I see a benefit to using
```js
const Promise = require('bluebird');

Promise.filter(
  functions.map(f => cloudWatchLogs.(...).promise(),
  filterFn
)
  .then(...)
```
over
```js
Promise.all(functions.map(f => cloudWatchLogs.(...).promise())
  .then(res => res.filter(filterFn))
  .then(...)
```

In fact, after writing it out, I think I prefer the chained filtering as it's much clearer that the filtering happens on the _resolved_ promises, something that wasn't immediately clear to me until reading the documentation.

**Pull request closed without merging**